graylog2-server/src/main/java/org/graylog2/messageprocessors/MessageProcessorModule.java

```diff
@@ -22,8 +22,8 @@
     @Override
     protected void configure() {
         addMessageProcessor(MessageFilterChainProcessor.class, MessageFilterChainProcessor.Descriptor.class);
-
-        bind(OrderedMessageProcessors.class).asEagerSingleton();
+        // must not be a singleton, because each thread should get an isolated copy of the processors
+        bind(OrderedMessageProcessors.class);
```

Comment:
Maybe make it very explicit here with `bind(OrderedMessageProcessors.class).in(Scopes.NO_SCOPE)`?